scheduler/src/cook/schema.clj

```diff
@@ -488,6 +494,34 @@ for a job. E.g. {:resources {:cpus 4 :mem 3} :constraints {\"unique_host_constra
     :db.install/_attribute :db.part/db}
    {:db/id (d/tempid :db.part/user)
     :db/ident :constraint.operator/equals}
+   ;; Checkpoint attributes
+   {:db/id (d/tempid :db.part/db)
+    :db/ident :checkpoint/mode
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one
+    :db.install/_attribute :db.part/db}
+   {:db/id (d/tempid :db.part/db)
+    :db/ident :checkpoint/options
+    :db/valueType :db.type/ref
+    :db/isComponent true
+    :db/cardinality :db.cardinality/one
+    :db.install/_attribute :db.part/db}
+   {:db/id (d/tempid :db.part/db)
+    :db/ident :checkpoint/periodic-options
+    :db/valueType :db.type/ref
+    :db/isComponent true
+    :db/cardinality :db.cardinality/one
+    :db.install/_attribute :db.part/db}
+   {:db/id (d/tempid :db.part/db)
+    :db/ident :checkpoint-options/preserve-paths
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/many
+    :db.install/_attribute :db.part/db}
+   {:db/id (d/tempid :db.part/db)
+    :db/ident :checkpoint-periodic-options/period-sec
```

Comment:
Does this automatically make it a sub-attribute of `:checkpoint/periodic-options`?

Reply:
this does
`{:db/id (d/tempid :db.part/db)
    :db/ident :checkpoint/periodic-options
    :db/valueType :db.type/ref
    :db/isComponent true
    :db/cardinality :db.cardinality/one
    :db.install/_attribute :db.part/db}`
the  :db.type/ref :db/isComponent